src/worker/worker/collectors/base_web_collector.py

```diff
@@ -58,7 +73,9 @@ def update_favicon(self, web_url: str, osint_source_id: str):
         self.core_api.update_osint_source_icon(osint_source_id, icon_content)
         return None
 
-    def web_content_from_article(self, web_url: str) -> tuple[str, datetime.datetime | None]:
+    def fetch_article_content(self, web_url: str, js_enabled: str = "false") -> tuple[str, datetime.datetime | None]:
+        if js_enabled == "true" or self.js_all == "true":
+            return self.fetch_content_with_js(web_url), None
```

Comment:
```
 def fetch_article_content(self, web_url: str) -> tuple[str, datetime.datetime | None]:
        if self.browser_mode:
                    return self.fetch_content_with_js(web_url), None
```

Reply:
wouldn't this lead to creating and closing the context for each URL call?